tox.ini

```diff
@@ -1,27 +1,32 @@
+# Tox configuration file
+# Read more under https://tox.readthedocs.org/
+
 [tox]
+minversion = 3.25
 envlist = tests
 skipsdist = True
+isolated_build = True
 
 [testenv]
+# shared conf for all testenvs
 basepython = python3.9
 usedevelop = True
-isolated_build = True
 setenv =
     TOXINIDIR = {toxinidir}
-    VIRTUALENV_PIP=21.3.1
+    VIRTUALENV_PIP=22.2
 
 [testenv:tests]
 description =
-    Invoke pytest to run automated tests.
+    Run automated tests.
 extras =
     testing
 commands =
-    python -m pytest tests --verbose --cov --cov-report term-missing --junitxml=test_report.xml --doctest-modules src
-    python -m pytest --pylint src/
-    python -m pytest --pylint tests/ --pylint-rcfile=tests/.pylintrc
-    #pytest --mypy src/ --verbose --strict
-    #pytest --mypy tests/ --verbose --strict
-    pytest --isort tests/ src/ --verbose
+    python --version
+    python -m pip --version
+    python -m pytest --verbose --doctest-modules --isort --pylint src
+    python -m pytest --verbose --cov --cov-report=term-missing --junitxml=test_report.xml --isort --doctest-modules --pylint --pylint-rcfile=tests/.pylintrc tests
+    #pytest --mypy src --verbose --strict
+    #pytest --mypy tests --verbose --strict
```

Comment:
Why do we keep commented out lines?

Reply:
The idea is to use mypy eventually, we just haven't yet gotten around to fixing the things it complains about.

Reply:
I'll make another MR about it.